sw/device/lib/dif/dif_edn.c

```diff
@@ -244,13 +244,12 @@ dif_result_t dif_edn_reseed(const dif_edn_t *edn,
   if (edn == NULL || seed_material == NULL) {
     return kDifBadArg;
   }
-  dif_csrng_seed_material_t seed_material2;
-  memcpy(&seed_material2, seed_material, sizeof(seed_material2));
-  return csrng_send_app_cmd(edn->base_addr, EDN_SW_CMD_REQ_REG_OFFSET,
-                            (csrng_app_cmd_t){
-                                .id = kCsrngAppCmdReseed,
-                                .seed_material = &seed_material2,
-                            });
+  return csrng_send_app_cmd(
+      edn->base_addr, EDN_SW_CMD_REQ_REG_OFFSET,
+      (csrng_app_cmd_t){
+          .id = kCsrngAppCmdReseed,
+          .seed_material = (const dif_csrng_seed_material_t *)seed_material,
```

Comment:
I just noticed that the `seed_material` has different types, and I think that the `memcpy` is there to avoid violating strict aliasing rules.
@luismarques is this cast safe?

Reply:
Interesting thought; I hadn't considered strict aliasing rules here. Could or should a `union` type help here?

Reply:
A `union` doesn't help